TopMovies/Service/MovieService.swift

```diff
@@ -35,7 +35,17 @@ class MovieService: StoreSubscriber {
                          $0.results.map { Movie(dto: $0) })}
             .forEach(mainStore.dispatch)
           mainStore.dispatch(MovieCategoriesAction
-                              .completed(categories: categoriesList.map { MovieCategory(dto: $0) }))
+                              .completed(
+                                categories: categoriesList.map { MovieCategory(dto: $0) },
+                                movies: categoriesList
+                                  .map { $0.results.map { Movie(dto: $0) } }
```

Comment:
I'd suggest storing movies in a dictionary inside of the event since you mix all movies from different categories in an array. If we fetch 50 items, most of them may be duplicated, but you will still append them to the array.